exports/taskfiles/utils/boost.yaml

```
@@ -0,0 +1,166 @@
+version: "3"
+
+includes:
+  remote: "remote.yaml"
+
+set: ["u", "pipefail"]
+shopt: ["globstar"]
+
+tasks:
+  # Runs the bootstrap.sh generate step in the given source directory. Boost only supports
+  # in-source generation and building.
+  #
+  # @param {string} SOURCE_DIR Project source directory.
+  # @param {string} INSTALL_PREFIX Path prefix of where the project should be installed.
+  # @param {string[]} TARGETS Target libraries to build.
+  # @param {string[]} [EXTRA_ARGS] Any additional arguments to pass to the generate command.
+  generate:
+    internal: true
+    dir: "{{.SOURCE_DIR}}"
+    cmds:
+      - >-
+        ./bootstrap.sh
+        --prefix="{{.INSTALL_PREFIX}}"
+        --exec-prefix="{{.INSTALL_PREFIX}}"
+        --with-libraries={{(join "," .TARGETS)}}
+        {{- range .EXTRA_ARGS}}
+        "{{.}}"
+        {{- end}}
+
+  # Runs the b2 build step for boost. The caller must have previously called `generate` on
+  # `SOURCE_DIR` for this task to succeed.
+  #
+  # @param {string} SOURCE_DIR Directory containing the boost source.
+  # @param {string[]} [EXTRA_ARGS] Any additional arguments to pass to the build command.
+  # @param {int} [JOBS] The maximum number of concurrent processes to use when building. If
+  # omitted, the b2 default number is used. Before 1.76.0, the number was 1. Since 1.76.0, the
+  # default is the number of cores.
+  build:
+    internal: true
+    dir: "{{.SOURCE_DIR}}"
+    cmds:
+      - >-
+        ./b2
+        {{- range .EXTRA_ARGS}}
+        "{{.}}"
+        {{- end}}
+        {{- if .JOBS}}
+        "-j{{.JOBS}}"
+        {{- end}}
+
+  # Runs the b2 install step for boost. The caller must have previously called `build` on
+  # `SOURCE_DIR` for this task to succeed. If `CMAKE_SETTINGS_DIR` is set, a settings file will be
+  # created in that directory, containing a `boost_ROOT` CMake variable that points to
+  # `INSTALL_PREFIX`.
+  #
+  # @param {string} SOURCE_DIR Directory containing the boost source.
+  # @param {string} INSTALL_PREFIX Path prefix of where the project should be installed.
+  # @param {string} [CMAKE_SETTINGS_DIR] If set, the directory where the project's CMake settings
+  # file should be stored.
+  # @param {string[]} [EXTRA_ARGS] Any additional arguments to pass to the install command.
+  install:
+    internal: true
+    dir: "{{.SOURCE_DIR}}"
+    cmds:
+      - >-
+        ./b2
+        install
+        {{- range .EXTRA_ARGS}}
+        "{{.}}"
+        {{- end}}
+      - >-
+        {{- if .CMAKE_SETTINGS_DIR}}
+          echo "set(BOOST_ROOT
```

Let's use `<PackageName_ROOT>` as it is supported much earlier (3.12) compared to `<PACKAGENAME_ROOT>` (3.27). This should avoid us needing to set policies in most projects.
See:
https://cmake.org/cmake/help/latest/policy/CMP0144.html
https://cmake.org/cmake/help/latest/envvar/PackageName_ROOT.html#envvar:%3CPackageName%3E_ROOT

```suggestion
          echo "set(Boost_ROOT
```